### PR TITLE
Added sorting by PK, then fields before properties, and fields/properties by name. Fixed tuple element name deduction and added tests for cross assembly tuple element name deduction.

### PR DESCRIPTION
**Pull request by skent-mz**

### CODE REVIEW

File: SQLite.Net.Tests/FieldSupportTest.cs [hunk 1]
```diff
@@ -27,6 +27,16 @@ public void AssertNotSet()
             Assert.That(shouldNotBeSet2, Is.EqualTo(-2));
         }
     }
+
+    public class FieldTestModelWithPkProperty
+    {
+        [PrimaryKey]
+        public int Id { get; set; }
```
Comment (johnfoconnor) [inline SQLite.Net.Tests/FieldSupportTest.cs:34]: can you move this to the last declaration position, will be a better unit test

File: src/SQLite.Net/Interop/ReflectionService.cs [hunk 2]
```diff
@@ -33,15 +33,45 @@ public static class ReflectionService
...
             var properties = mappedType.GetTypeInfo().GetRuntimeProperties()
                 .Where(p => p.CanRead && p.CanWrite && p.GetMethod.IsPublic && p.SetMethod.IsPublic &&
                             !p.GetMethod.IsStatic && !p.SetMethod.IsStatic);
             var fields = mappedType.GetTypeInfo().GetRuntimeFields()
                 .Where(f => f.IsPublic && !f.IsStatic && !f.IsInitOnly);
 
-            return fields.Union<MemberInfo>(properties).ToList();
+            var members = fields
+                .Union<MemberInfo>(properties).ToList();
+            members.Sort((l, r) =>
```
Comment (johnfoconnor) [inline src/SQLite.Net/Interop/ReflectionService.cs:46]: can you comment and summarize the sorting rules/behavior here

**Review by johnfoconnor — APPROVED** [review]

File: src/SQLite.Net/Interop/ReflectionService.cs (hunk 2 above)
Comment (skent-mz) [reply src/SQLite.Net/Interop/ReflectionService.cs:46]: Resolved in PR #10

File: SQLite.Net.Tests/FieldSupportTest.cs (hunk 1 above)
Comment (skent-mz) [reply SQLite.Net.Tests/FieldSupportTest.cs:34]: Resolved in PR #10

**Pull request merged**